src/epochs.lua

```diff
@@ -722,52 +722,53 @@ function epochs.distributeRewardsForEpoch(currentTimestamp)
 					/ totalEligibleRewardsForGatewayAndDelegates -- percent of what was earned vs what was eligible
 				-- optimally this is 1, but if the gateway did not do what it was supposed to do, it will be less than 1 and thus all payouts will be less
 				local totalDistributedToDelegates = 0
+				local totalRewardsForMissingDelegates = 0
 				-- distribute all the predetermined rewards to the delegates
 				for delegateAddress, eligibleDelegateReward in pairs(totalEligibleRewardsForGateway.delegateRewards) do
 					local actualDelegateReward = math.floor(eligibleDelegateReward * percentOfEligibleEarned)
-					-- distribute the rewards to the delegate if greater than 0
+					-- distribute the rewards to the delegate if greater than 0 and the delegate still exists on the gateway
 					if actualDelegateReward > 0 then
-						-- increase the stake and decrease the protocol balance, returns the updated gateway
-						gateway = gar.increaseExistingDelegateStake(
-							gatewayAddress,
-							gateway,
-							delegateAddress,
-							actualDelegateReward
-						)
-						balances.reduceBalance(ao.id, actualDelegateReward)
+						if gateway.delegates[delegateAddress] then
+							-- increase the stake and decrease the protocol balance, returns the updated gateway
+							gateway = gar.increaseExistingDelegateStake(
+								gatewayAddress,
+								gateway,
+								delegateAddress,
+								actualDelegateReward
+							)
+							balances.reduceBalance(ao.id, actualDelegateReward)
+							-- update the distributed rewards for the delegate
+							distributed[delegateAddress] = (distributed[delegateAddress] or 0) + actualDelegateReward
+							totalDistributedToDelegates = totalDistributedToDelegates + actualDelegateReward
+						else
+							totalRewardsForMissingDelegates = totalRewardsForMissingDelegates + actualDelegateReward
+						end
 					end
-					-- increment the total distributed
-					totalDistributed = math.floor(totalDistributed + actualDelegateReward)
-					-- update the distributed rewards for the delegate
-					distributed[delegateAddress] = (distributed[delegateAddress] or 0) + actualDelegateReward
-					-- increment the total distributed for the epoch
-					totalDistributedToDelegates = totalDistributedToDelegates + actualDelegateReward
 				end
 				-- transfer the remaining rewards to the gateway
-				local actualOperatorReward =
-					math.floor(earnedRewardForGatewayAndDelegates - totalDistributedToDelegates)
+				local actualOperatorReward = math.floor(
+					earnedRewardForGatewayAndDelegates - totalDistributedToDelegates - totalRewardsForMissingDelegates
```

Comment:
So the missing delegate stake just goes back to the gateway since it'd be challenging to go back through the delegates list and give each their fair share of the total not distributed to the missing delegates?

Reply:
no - missing delegate doesn't get distributed at all. This step accounts for rounding, returning any extra stake that should have been distributed back to the gateway.